usaspending_api/financial_activities/models.py

```diff
@@ -56,3 +56,165 @@ class FinancialAccountsByProgramActivityObjectClass(DataSourceTrackedModel):
+            current.gross_outlay_amount_by_program_object_class_fyb - COALESCE(previous.gross_outlay_amount_by_program_object_class_fyb, 0) AS gross_outlay_amount_by_program_object_class_fyb,
+            current.gross_outlay_amount_by_program_object_class_cpe - COALESCE(previous.gross_outlay_amount_by_program_object_class_cpe, 0) AS gross_outlay_amount_by_program_object_class_cpe,
+            current.obligations_incurred_by_program_object_class_cpe - COALESCE(previous.obligations_incurred_by_program_object_class_cpe, 0) AS obligations_incurred_by_program_object_class_cpe,
+            current.ussgl487100_down_adj_pri_unpaid_undel_orders_oblig_recov_cpe - COALESCE(previous.ussgl487100_down_adj_pri_unpaid_undel_orders_oblig_recov_cpe, 0) AS ussgl487100_down_adj_pri_unpaid_undel_orders_oblig_recov_cpe,
+            current.ussgl497100_down_adj_pri_unpaid_deliv_orders_oblig_recov_cpe - COALESCE(previous.ussgl497100_down_adj_pri_unpaid_deliv_orders_oblig_recov_cpe, 0) AS ussgl497100_down_adj_pri_unpaid_deliv_orders_oblig_recov_cpe,
+            current.ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe - COALESCE(previous.ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe, 0) AS ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe,
+            current.ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe - COALESCE(previous.ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe, 0) AS ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe,
+            current.deobligations_recoveries_refund_pri_program_object_class_cpe - COALESCE(previous.deobligations_recoveries_refund_pri_program_object_class_cpe, 0) AS deobligations_recoveries_refund_pri_program_object_class_cpe
+        FROM
+            financial_accounts_by_program_activity_object_class AS current
+            JOIN submission_attributes AS sub
+            ON current.submission_id = sub.submission_id
+            LEFT JOIN financial_accounts_by_program_activity_object_class AS previous
+            ON current.treasury_account_id = previous.treasury_account_id
+            AND current.object_class_id = previous.object_class_id
+            AND current.program_activity_id = previous.program_activity_id
+            AND previous.submission_id = sub.previous_submission_id
+    """
+
+    @classmethod
+    def get_quarterly_numbers(cls, current_submission_id=None):
+        """
+        Return a RawQuerySet of quarterly financial numbers by TAS,
+        object class, and program activity.
+        """
+        if current_submission_id is None:
+            return FinancialAccountsByProgramActivityObjectClass.objects.raw(
+                cls.QUARTERLY_SQL)
+        else:
+            sql = cls.QUARTERLY_SQL + ' WHERE current.submission_id = %s'
+            return FinancialAccountsByProgramActivityObjectClass.objects.raw(
+                sql, [current_submission_id])
+
+
+class TasProgramActivityObjectClassQuarterly(DataSourceTrackedModel):
```

Comment:
Yes, great idea! At the moment, the fields between these two models aren't exactly the same, since the old one has a bunch of unused columns (almost removed those as part of this PR, but decided not to pull this particular thread right now).

I added a general "accounts financial models" cleanup story: https://federal-spending-transparency.atlassian.net/browse/DS-1026

Thanks for the prompt--we should absolutely do this.